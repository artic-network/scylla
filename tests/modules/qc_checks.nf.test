nextflow_workflow {

    name "qc_checks module tests"
    script "modules/qc_checks.nf"
    workflow "qc_checks"
    tag "unit"
    tag "ci"

    test("qc_check_success") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            workflow {
                """
                input[0] = Channel.of(
                    tuple(
                        "test",
                        file("${projectDir}/tests/test-data/barcode01/barcode01.fq.gz", checkIfExists: true)
                    )
                )
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out, path("${outputDir}/test/qc/total_length.json")).match()
        }

    }

}
