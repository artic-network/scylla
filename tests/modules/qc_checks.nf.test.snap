{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        "test",
                        "raw_reads.stats:md5,e6987c5ac7822172c74d9b7c97383aca"
                    ]
                ]
            },
            "total_length.json:md5,ea9aaf83c95f9c31ca20720e51ee4fd0"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-27T14:05:05.540986896"
    }
}